\hypertarget{classv8_1_1TryCatch}{}\doxysection{v8\+::Try\+Catch Class Reference}
\label{classv8_1_1TryCatch}\index{v8::TryCatch@{v8::TryCatch}}


An external exception handler.  




{\ttfamily \#include $<$v8-\/exception.\+h$>$}



Collaboration diagram for v8\+::Try\+Catch\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch__coll__graph}
\end{center}
\end{figure}
\doxysubsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{classv8_1_1TryCatch_a623ce624491b383fae60c05d2aeefb1a}{Try\+Catch}} (\mbox{\hyperlink{classv8_1_1Isolate}{Isolate}} $\ast$isolate)
\begin{DoxyCompactList}\small\item\em Creates a new try/catch block and registers it with \mbox{\hyperlink{namespacev8}{v8}}. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classv8_1_1TryCatch_a2c9ad4b40d17dd31c6dd020736b30679}{$\sim$\+Try\+Catch}} ()
\begin{DoxyCompactList}\small\item\em Unregisters and deletes this try/catch block. \end{DoxyCompactList}\item 
bool \mbox{\hyperlink{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e}{Has\+Caught}} () const
\begin{DoxyCompactList}\small\item\em Returns true if an exception has been caught by this try/catch block. \end{DoxyCompactList}\item 
bool \mbox{\hyperlink{classv8_1_1TryCatch_ae94d0f0d695dbfecbfcdeb48716b367f}{Can\+Continue}} () const
\begin{DoxyCompactList}\small\item\em For certain types of exceptions, it makes no sense to continue execution. \end{DoxyCompactList}\item 
bool \mbox{\hyperlink{classv8_1_1TryCatch_a3175391e782afc092495616b4d35b91b}{Has\+Terminated}} () const
\begin{DoxyCompactList}\small\item\em Returns true if an exception has been caught due to script execution being terminated. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classv8_1_1Local}{Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{Value}} $>$ \mbox{\hyperlink{classv8_1_1TryCatch_ab2cbdeeea106bcda2026b980e94c29ba}{Re\+Throw}} ()
\begin{DoxyCompactList}\small\item\em Throws the exception caught by this \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} in a way that avoids it being caught again by this same \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}}. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classv8_1_1Local}{Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{Value}} $>$ \mbox{\hyperlink{classv8_1_1TryCatch_a5b86f4229e0725a3eacc69826da2e984}{Exception}} () const
\begin{DoxyCompactList}\small\item\em Returns the exception caught by this try/catch block. \end{DoxyCompactList}\item 
\mbox{\hyperlink{v8config_8h_a63258e4bf5dc9fb8c78f881c8405d83b}{V8\+\_\+\+WARN\+\_\+\+UNUSED\+\_\+\+RESULT}} \mbox{\hyperlink{classv8_1_1MaybeLocal}{Maybe\+Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{Value}} $>$ \mbox{\hyperlink{classv8_1_1TryCatch_a40e58ea4e31de973e22b2c9d86d30d57}{Stack\+Trace}} (\mbox{\hyperlink{classv8_1_1Local}{Local}}$<$ \mbox{\hyperlink{classv8_1_1Context}{Context}} $>$ context) const
\begin{DoxyCompactList}\small\item\em Returns the .stack property of the thrown object. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classv8_1_1Local}{Local}}$<$ \mbox{\hyperlink{classv8_1_1Message}{v8\+::\+Message}} $>$ \mbox{\hyperlink{classv8_1_1TryCatch_a05651b0b264e592de56f5e16bc80057f}{Message}} () const
\begin{DoxyCompactList}\small\item\em Returns the message associated with this exception. \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classv8_1_1TryCatch_a3aae8acab4c99b374b7d782763d4c8e1}{Reset}} ()
\begin{DoxyCompactList}\small\item\em Clears any exceptions that may have been caught by this try/catch block. \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classv8_1_1TryCatch_a032cd889d76bd596e2616df11ced8682}{Set\+Verbose}} (bool value)
\begin{DoxyCompactList}\small\item\em \mbox{\hyperlink{classv8_1_1Set}{Set}} verbosity of the external exception handler. \end{DoxyCompactList}\item 
bool \mbox{\hyperlink{classv8_1_1TryCatch_aa4e77abd0d6790700442153cb036517f}{Is\+Verbose}} () const
\begin{DoxyCompactList}\small\item\em Returns true if verbosity is enabled. \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classv8_1_1TryCatch_a541b8fa6951bd5a439692c22d5c7b73c}{Set\+Capture\+Message}} (bool value)
\begin{DoxyCompactList}\small\item\em \mbox{\hyperlink{classv8_1_1Set}{Set}} whether or not this \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} should capture a \mbox{\hyperlink{classv8_1_1Message}{Message}} object which holds source information about where the exception occurred. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classv8_1_1TryCatch_a7368ba9d21e9bddef45a76f537f1900e}{Try\+Catch}} (const \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} \&)=delete
\item 
void \mbox{\hyperlink{classv8_1_1TryCatch_a9fec223509a6d7095b3088871a6084ce}{operator=}} (const \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} \&)=delete
\end{DoxyCompactItemize}
\doxysubsection*{Static Public Member Functions}
\begin{DoxyCompactItemize}
\item 
static \mbox{\hyperlink{v8config_8h_a63258e4bf5dc9fb8c78f881c8405d83b}{V8\+\_\+\+WARN\+\_\+\+UNUSED\+\_\+\+RESULT}} \mbox{\hyperlink{classv8_1_1MaybeLocal}{Maybe\+Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{Value}} $>$ \mbox{\hyperlink{classv8_1_1TryCatch_a5d80d54b34da0973f1acb30857add186}{Stack\+Trace}} (\mbox{\hyperlink{classv8_1_1Local}{Local}}$<$ \mbox{\hyperlink{classv8_1_1Context}{Context}} $>$ context, \mbox{\hyperlink{classv8_1_1Local}{Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{Value}} $>$ exception)
\begin{DoxyCompactList}\small\item\em Returns the .stack property of an object. \end{DoxyCompactList}\end{DoxyCompactItemize}
\doxysubsection*{Private Member Functions}
\begin{DoxyCompactItemize}
\item 
void $\ast$ \mbox{\hyperlink{classv8_1_1TryCatch_aa6b27cdc25a9ce0806144338abd224e0}{operator new}} (size\+\_\+t \mbox{\hyperlink{flag-definitions_8h_a3df0b28b666e942ffb9d7d04acdb9715}{size}})
\item 
void $\ast$ \mbox{\hyperlink{classv8_1_1TryCatch_a9a9c35201c272ed3a0f9d9d6526c8637}{operator new\mbox{[}$\,$\mbox{]}}} (size\+\_\+t \mbox{\hyperlink{flag-definitions_8h_a3df0b28b666e942ffb9d7d04acdb9715}{size}})
\item 
void \mbox{\hyperlink{classv8_1_1TryCatch_ad400a5dc4796e85d3d97a0e612adf0f8}{operator delete}} (void $\ast$, size\+\_\+t)
\item 
void \mbox{\hyperlink{classv8_1_1TryCatch_aebc50debeedcab35a6a8396f73bd122a}{operator delete\mbox{[}$\,$\mbox{]}}} (void $\ast$, size\+\_\+t)
\item 
\mbox{\hyperlink{namespacev8_1_1internal_ab4955debfb9384e59e3d4e9fcab5b0a0}{internal\+::\+Address}} \mbox{\hyperlink{classv8_1_1TryCatch_a8aab3d2c4dabe7ea5ffc88ab22fbdc46}{JSStack\+Comparable\+Address\+Private}} ()
\begin{DoxyCompactList}\small\item\em There are cases when the raw address of C++ \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} object cannot be used for comparisons with addresses into the JS stack. \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classv8_1_1TryCatch_ae0c8d52ffa44aaae9f9ed6aed8571aa6}{Reset\+Internal}} ()
\end{DoxyCompactItemize}
\doxysubsection*{Private Attributes}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{classv8_1_1internal_1_1Isolate}{internal\+::\+Isolate}} $\ast$ \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\+\_\+isolate\+\_\+}}
\item 
\mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} $\ast$ \mbox{\hyperlink{classv8_1_1TryCatch_ae1e039966ea18612bb2ecfdd2602ea84}{next\+\_\+}}
\item 
void $\ast$ \mbox{\hyperlink{classv8_1_1TryCatch_a4ae6a166af0336716516addc1f64fe1f}{exception\+\_\+}}
\item 
void $\ast$ \mbox{\hyperlink{classv8_1_1TryCatch_a6efc81ccc783f40c660f0e709b578aaa}{message\+\_\+obj\+\_\+}}
\item 
\mbox{\hyperlink{namespacev8_1_1internal_ab4955debfb9384e59e3d4e9fcab5b0a0}{internal\+::\+Address}} \mbox{\hyperlink{classv8_1_1TryCatch_a9c7affe9ed0b8faf49573a0709264039}{js\+\_\+stack\+\_\+comparable\+\_\+address\+\_\+}}
\item 
bool \mbox{\hyperlink{classv8_1_1TryCatch_a3c2798d8716a14df4550bfadefdcf6a7}{is\+\_\+verbose\+\_\+}}\+: 1
\item 
bool \mbox{\hyperlink{classv8_1_1TryCatch_a92c1bd92eecbc728d6c2150c338fef7d}{can\+\_\+continue\+\_\+}}\+: 1
\item 
bool \mbox{\hyperlink{classv8_1_1TryCatch_ad1f5b0368678fa2bd0d73ac17acf976e}{capture\+\_\+message\+\_\+}}\+: 1
\item 
bool \mbox{\hyperlink{classv8_1_1TryCatch_a6c8efeb878409539d0863314fba98a23}{rethrow\+\_\+}}\+: 1
\end{DoxyCompactItemize}
\doxysubsection*{Friends}
\begin{DoxyCompactItemize}
\item 
class \mbox{\hyperlink{classv8_1_1TryCatch_a438c8b784134d31d7a50bc60098d8ba1}{internal\+::\+Isolate}}
\item 
class \mbox{\hyperlink{classv8_1_1TryCatch_a97fa4264c507616255cedf7a87ec1910}{internal\+::\+Thread\+Local\+Top}}
\end{DoxyCompactItemize}


\doxysubsection{Detailed Description}
An external exception handler. 

Definition at line 138 of file v8-\/exception.\+h.



\doxysubsection{Constructor \& Destructor Documentation}
\mbox{\Hypertarget{classv8_1_1TryCatch_a623ce624491b383fae60c05d2aeefb1a}\label{classv8_1_1TryCatch_a623ce624491b383fae60c05d2aeefb1a}} 
\index{v8::TryCatch@{v8::TryCatch}!TryCatch@{TryCatch}}
\index{TryCatch@{TryCatch}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{TryCatch()}{TryCatch()}\hspace{0.1cm}{\footnotesize\ttfamily [1/2]}}
{\footnotesize\ttfamily v8\+::\+Try\+Catch\+::\+Try\+Catch (\begin{DoxyParamCaption}\item[{\mbox{\hyperlink{classv8_1_1Isolate}{v8\+::\+Isolate}} $\ast$}]{v8\+\_\+isolate }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [explicit]}}



Creates a new try/catch block and registers it with \mbox{\hyperlink{namespacev8}{v8}}. 

Note that all \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} blocks should be stack allocated because the memory location itself is compared against Java\+Script try/catch blocks. 

Definition at line 2772 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2773     : \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}(\textcolor{keyword}{reinterpret\_cast<}\mbox{\hyperlink{classv8_1_1internal_1_1Isolate}{i::Isolate}}*\textcolor{keyword}{>}(v8\_isolate)),}
\DoxyCodeLine{2774       \mbox{\hyperlink{classv8_1_1TryCatch_ae1e039966ea18612bb2ecfdd2602ea84}{next\_}}(\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_a94a97108717e97fa3c032710ca045bdf}{try\_catch\_handler}}()),}
\DoxyCodeLine{2775       \mbox{\hyperlink{classv8_1_1TryCatch_a3c2798d8716a14df4550bfadefdcf6a7}{is\_verbose\_}}(\textcolor{keyword}{false}),}
\DoxyCodeLine{2776       \mbox{\hyperlink{classv8_1_1TryCatch_a92c1bd92eecbc728d6c2150c338fef7d}{can\_continue\_}}(\textcolor{keyword}{true}),}
\DoxyCodeLine{2777       \mbox{\hyperlink{classv8_1_1TryCatch_ad1f5b0368678fa2bd0d73ac17acf976e}{capture\_message\_}}(\textcolor{keyword}{true}),}
\DoxyCodeLine{2778       \mbox{\hyperlink{classv8_1_1TryCatch_a6c8efeb878409539d0863314fba98a23}{rethrow\_}}(\textcolor{keyword}{false}) \{}
\DoxyCodeLine{2779   \mbox{\hyperlink{classv8_1_1TryCatch_ae0c8d52ffa44aaae9f9ed6aed8571aa6}{ResetInternal}}();}
\DoxyCodeLine{2780   \textcolor{comment}{// Special handling for simulators which have a separate JS stack.}}
\DoxyCodeLine{2781   \mbox{\hyperlink{classv8_1_1TryCatch_a9c7affe9ed0b8faf49573a0709264039}{js\_stack\_comparable\_address\_}} = \textcolor{keyword}{static\_cast<}\mbox{\hyperlink{namespacev8_1_1internal_ab4955debfb9384e59e3d4e9fcab5b0a0}{internal::Address}}\textcolor{keyword}{>}(}
\DoxyCodeLine{2782       \mbox{\hyperlink{classv8_1_1internal_1_1SimulatorStack_afce69f26f86d096825381c229a09155f}{i::SimulatorStack::RegisterJSStackComparableAddress}}(\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}));}
\DoxyCodeLine{2783   \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_af44fbb26816a718884f8e54a6e055c5e}{RegisterTryCatchHandler}}(\textcolor{keyword}{this});}
\DoxyCodeLine{2784 \}}

\end{DoxyCode}


References i\+\_\+isolate\+\_\+, js\+\_\+stack\+\_\+comparable\+\_\+address\+\_\+, v8\+::internal\+::\+Simulator\+Stack\+::\+Register\+JSStack\+Comparable\+Address(), v8\+::internal\+::\+Isolate\+::\+Register\+Try\+Catch\+Handler(), and Reset\+Internal().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a623ce624491b383fae60c05d2aeefb1a_cgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a2c9ad4b40d17dd31c6dd020736b30679}\label{classv8_1_1TryCatch_a2c9ad4b40d17dd31c6dd020736b30679}} 
\index{v8::TryCatch@{v8::TryCatch}!````~TryCatch@{$\sim$TryCatch}}
\index{````~TryCatch@{$\sim$TryCatch}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{$\sim$TryCatch()}{~TryCatch()}}
{\footnotesize\ttfamily v8\+::\+Try\+Catch\+::$\sim$\+Try\+Catch (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})}



Unregisters and deletes this try/catch block. 



Definition at line 2794 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2794                       \{}
\DoxyCodeLine{2795   \textcolor{keywordflow}{if} (\mbox{\hyperlink{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e}{HasCaught}}()) \{}
\DoxyCodeLine{2796     \textcolor{keywordflow}{if} (\mbox{\hyperlink{classv8_1_1TryCatch_a6c8efeb878409539d0863314fba98a23}{rethrow\_}} || (\mbox{\hyperlink{v8config_8h_a67bd29db07c711b0fb11ff1c324ebb60}{V8\_UNLIKELY}}(\mbox{\hyperlink{classv8_1_1TryCatch_a3175391e782afc092495616b4d35b91b}{HasTerminated}}()) \&\&}
\DoxyCodeLine{2797                      !\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_aa3f53c9b9733db23efd7d26d2c854409}{thread\_local\_top}}()-\/>\mbox{\hyperlink{classv8_1_1internal_1_1ThreadLocalTop_a59794f98631d9b7aaaaf7adac9fa7fba}{CallDepthIsZero}}())) \{}
\DoxyCodeLine{2798       \textcolor{keywordflow}{if} (\mbox{\hyperlink{classv8_1_1TryCatch_ad1f5b0368678fa2bd0d73ac17acf976e}{capture\_message\_}}) \{}
\DoxyCodeLine{2799         \textcolor{comment}{// If an exception was caught and rethrow\_ is indicated, the saved}}
\DoxyCodeLine{2800         \textcolor{comment}{// message, script, and location need to be restored to Isolate TLS}}
\DoxyCodeLine{2801         \textcolor{comment}{// for reuse.  capture\_message\_ needs to be disabled so that Throw()}}
\DoxyCodeLine{2802         \textcolor{comment}{// does not create a new message.}}
\DoxyCodeLine{2803         \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_aa3f53c9b9733db23efd7d26d2c854409}{thread\_local\_top}}()-\/>\mbox{\hyperlink{classv8_1_1internal_1_1ThreadLocalTop_ac767605ceac45f768c206809d0152aba}{rethrowing\_message\_}} = \textcolor{keyword}{true};}
\DoxyCodeLine{2804         \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_a45d65d87a2bf3a5417bb35140951f885}{set\_pending\_message}}(\mbox{\hyperlink{namespacev8_1_1anonymous__namespace_02api_8cc_03_ae7165265166edb47786472c2cf9a92de}{ToObject}}(\mbox{\hyperlink{classv8_1_1TryCatch_a6efc81ccc783f40c660f0e709b578aaa}{message\_obj\_}}));}
\DoxyCodeLine{2805       \}}
\DoxyCodeLine{2806       \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_a54758a65365a4c4f48a945100b647584}{UnregisterTryCatchHandler}}(\textcolor{keyword}{this});}
\DoxyCodeLine{2807       \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_af2996b3f59b1763c3036a36fb0405b14}{clear\_internal\_exception}}();}
\DoxyCodeLine{2808       \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_a5819188db3266ad85f50d20a1e254459}{Throw}}(\mbox{\hyperlink{namespacev8_1_1anonymous__namespace_02api_8cc_03_ae7165265166edb47786472c2cf9a92de}{ToObject}}(\mbox{\hyperlink{classv8_1_1TryCatch_a4ae6a166af0336716516addc1f64fe1f}{exception\_}}));}
\DoxyCodeLine{2809       \textcolor{keywordflow}{return};}
\DoxyCodeLine{2810     \}}
\DoxyCodeLine{2811     \mbox{\hyperlink{classv8_1_1TryCatch_a3aae8acab4c99b374b7d782763d4c8e1}{Reset}}();}
\DoxyCodeLine{2812   \}}
\DoxyCodeLine{2813   \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_a54758a65365a4c4f48a945100b647584}{UnregisterTryCatchHandler}}(\textcolor{keyword}{this});}
\DoxyCodeLine{2814   \mbox{\hyperlink{src_2base_2logging_8h_a96a75539aa82216cdde2250a11126ea0}{DCHECK\_IMPLIES}}(\mbox{\hyperlink{classv8_1_1TryCatch_a6c8efeb878409539d0863314fba98a23}{rethrow\_}},}
\DoxyCodeLine{2815                  !\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_aa3f53c9b9733db23efd7d26d2c854409}{thread\_local\_top}}()-\/>\mbox{\hyperlink{classv8_1_1internal_1_1ThreadLocalTop_ac767605ceac45f768c206809d0152aba}{rethrowing\_message\_}});}
\DoxyCodeLine{2816 \}}

\end{DoxyCode}


References DCHECK\+\_\+\+IMPLIES, v8\+::anonymous\+\_\+namespace\{api.\+cc\}\+::\+To\+Object(), and V8\+\_\+\+UNLIKELY.

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a2c9ad4b40d17dd31c6dd020736b30679_cgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a7368ba9d21e9bddef45a76f537f1900e}\label{classv8_1_1TryCatch_a7368ba9d21e9bddef45a76f537f1900e}} 
\index{v8::TryCatch@{v8::TryCatch}!TryCatch@{TryCatch}}
\index{TryCatch@{TryCatch}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{TryCatch()}{TryCatch()}\hspace{0.1cm}{\footnotesize\ttfamily [2/2]}}
{\footnotesize\ttfamily v8\+::\+Try\+Catch\+::\+Try\+Catch (\begin{DoxyParamCaption}\item[{const \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} \&}]{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [delete]}}



\doxysubsection{Member Function Documentation}
\mbox{\Hypertarget{classv8_1_1TryCatch_ae94d0f0d695dbfecbfcdeb48716b367f}\label{classv8_1_1TryCatch_ae94d0f0d695dbfecbfcdeb48716b367f}} 
\index{v8::TryCatch@{v8::TryCatch}!CanContinue@{CanContinue}}
\index{CanContinue@{CanContinue}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{CanContinue()}{CanContinue()}}
{\footnotesize\ttfamily bool v8\+::\+Try\+Catch\+::\+Can\+Continue (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const}



For certain types of exceptions, it makes no sense to continue execution. 

If Can\+Continue returns false, the correct action is to perform any C++ cleanup needed and then return. If Can\+Continue returns false and Has\+Terminated returns true, it is possible to call Cancel\+Terminate\+Execution in order to continue calling into the engine. 

Definition at line 2827 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2827 \{ \textcolor{keywordflow}{return} \mbox{\hyperlink{classv8_1_1TryCatch_a92c1bd92eecbc728d6c2150c338fef7d}{can\_continue\_}}; \}}

\end{DoxyCode}


Referenced by v8\+\_\+inspector\+::\+Injected\+Script\+::wrap\+Evaluate\+Result().

Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_ae94d0f0d695dbfecbfcdeb48716b367f_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a5b86f4229e0725a3eacc69826da2e984}\label{classv8_1_1TryCatch_a5b86f4229e0725a3eacc69826da2e984}} 
\index{v8::TryCatch@{v8::TryCatch}!Exception@{Exception}}
\index{Exception@{Exception}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{Exception()}{Exception()}}
{\footnotesize\ttfamily \mbox{\hyperlink{classv8_1_1Local}{v8\+::\+Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{Value}} $>$ v8\+::\+Try\+Catch\+::\+Exception (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const}



Returns the exception caught by this try/catch block. 

If no exception has been caught an empty handle is returned. 

Definition at line 2840 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2840                                            \{}
\DoxyCodeLine{2841   \textcolor{keywordflow}{if} (!\mbox{\hyperlink{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e}{HasCaught}}()) \textcolor{keywordflow}{return} \{\};}
\DoxyCodeLine{2842   \textcolor{keywordflow}{if} (\mbox{\hyperlink{classv8_1_1TryCatch_a3175391e782afc092495616b4d35b91b}{HasTerminated}}()) \{}
\DoxyCodeLine{2843     \textcolor{keywordflow}{return} v8::Utils::ToLocal(\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_a7e2962359009581ba84e6a361f574f61}{factory}}()-\/>null\_value());}
\DoxyCodeLine{2844   \}}
\DoxyCodeLine{2845   \textcolor{keywordflow}{return} v8::Utils::ToLocal(\mbox{\hyperlink{namespacev8_1_1internal_a73502f55102d18cb764881bc29f35c05}{i::direct\_handle}}(\mbox{\hyperlink{namespacev8_1_1anonymous__namespace_02api_8cc_03_ae7165265166edb47786472c2cf9a92de}{ToObject}}(\mbox{\hyperlink{classv8_1_1TryCatch_a4ae6a166af0336716516addc1f64fe1f}{exception\_}}), \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}));}
\DoxyCodeLine{2846 \}}

\end{DoxyCode}


References v8\+::internal\+::direct\+\_\+handle(), and v8\+::anonymous\+\_\+namespace\{api.\+cc\}\+::\+To\+Object().



Referenced by v8\+\_\+inspector\+::\+Injected\+Script\+::create\+Exception\+Details(), v8\+\_\+inspector\+::\+V8\+Inspector\+Session\+Impl\+::evaluate(), v8\+::\+Shell\+::\+Execute\+Module(), v8\+\_\+inspector\+::\+Value\+Mirror\+::get\+Properties(), v8\+::internal\+::anonymous\+\_\+namespace\{execution.\+cc\}\+::\+Invoke\+With\+Try\+Catch(), v8\+::anonymous\+\_\+namespace\{d8.\+cc\}\+::\+Reject\+Promise\+If\+Execution\+Is\+Not\+Terminating(), v8\+::\+Shell\+::\+Report\+Exception(), v8\+::\+Inspector\+Frontend\+::\+Send(), and v8\+\_\+inspector\+::\+Injected\+Script\+::wrap\+Evaluate\+Result().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a5b86f4229e0725a3eacc69826da2e984_cgraph}
\end{center}
\end{figure}
Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a5b86f4229e0725a3eacc69826da2e984_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e}\label{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e}} 
\index{v8::TryCatch@{v8::TryCatch}!HasCaught@{HasCaught}}
\index{HasCaught@{HasCaught}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{HasCaught()}{HasCaught()}}
{\footnotesize\ttfamily bool v8\+::\+Try\+Catch\+::\+Has\+Caught (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const}



Returns true if an exception has been caught by this try/catch block. 



Definition at line 2823 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2823                                  \{}
\DoxyCodeLine{2824   \textcolor{keywordflow}{return} !IsTheHole(\mbox{\hyperlink{namespacev8_1_1anonymous__namespace_02api_8cc_03_ae7165265166edb47786472c2cf9a92de}{ToObject}}(\mbox{\hyperlink{classv8_1_1TryCatch_a4ae6a166af0336716516addc1f64fe1f}{exception\_}}), \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}});}
\DoxyCodeLine{2825 \}}

\end{DoxyCode}


References v8\+::anonymous\+\_\+namespace\{api.\+cc\}\+::\+To\+Object().



Referenced by v8\+\_\+inspector\+::\+V8\+Runtime\+Agent\+Impl\+::compile\+Script(), v8\+\_\+inspector\+::\+Injected\+Script\+::create\+Exception\+Details(), v8\+::\+Shell\+::\+Create\+Realm(), v8\+\_\+inspector\+::anonymous\+\_\+namespace\{value-\/mirror.\+cc\}\+::does\+Attribute\+Have\+Observable\+Side\+Effect\+On\+Get(), v8\+\_\+inspector\+::\+V8\+Runtime\+Agent\+Impl\+::evaluate(), v8\+::internal\+::\+Source\+Text\+Module\+::\+Evaluate(), v8\+\_\+inspector\+::\+V8\+Inspector\+Session\+Impl\+::evaluate(), v8\+::\+Shell\+::\+Execute\+Module(), v8\+::\+Shell\+::\+Execute\+String(), v8\+\_\+inspector\+::\+Value\+Mirror\+::get\+Properties(), v8\+::\+Isolate\+::\+Has\+Pending\+Exception(), v8\+\_\+inspector\+::anonymous\+\_\+namespace\{v8-\/runtime-\/agent-\/impl.\+cc\}\+::inner\+Call\+Function\+On(), v8\+::internal\+::anonymous\+\_\+namespace\{execution.\+cc\}\+::\+Invoke\+With\+Try\+Catch(), v8\+::\+Shell\+::\+JSONModule\+Evaluation\+Steps(), v8\+::\+Shell\+::\+Load\+JSON(), v8\+::internal\+::anonymous\+\_\+namespace\{gc-\/extension.\+cc\}\+::\+Parse(), v8\+::internal\+::anonymous\+\_\+namespace\{liveedit.\+cc\}\+::\+Parse\+Script(), v8\+::\+Worker\+::\+Process\+Messages(), v8\+::anonymous\+\_\+namespace\{d8.\+cc\}\+::\+Process\+Messages(), v8\+\_\+inspector\+::\+V8\+Console\+::query\+Objects\+Callback(), v8\+::anonymous\+\_\+namespace\{d8.\+cc\}\+::\+Reject\+Promise\+If\+Execution\+Is\+Not\+Terminating(), v8\+\_\+inspector\+::anonymous\+\_\+namespace\{custom-\/preview.\+cc\}\+::report\+Error(), v8\+::internal\+::anonymous\+\_\+namespace\{snapshot.\+cc\}\+::\+Run\+Extra\+Code(), v8\+::\+Inspector\+Client\+::run\+Message\+Loop\+On\+Pause(), v8\+\_\+inspector\+::\+V8\+Runtime\+Agent\+Impl\+::run\+Script(), v8\+::\+Inspector\+Frontend\+::\+Send(), v8\+\_\+inspector\+::\+V8\+Debugger\+Agent\+Impl\+::set\+Variable\+Value(), v8\+::\+Async\+Hooks\+::\+Shell\+Promise\+Hook(), v8\+\_\+inspector\+::\+V8\+Regex\+::\+V8\+Regex(), and v8\+\_\+inspector\+::\+Injected\+Script\+::wrap\+Evaluate\+Result().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e_cgraph}
\end{center}
\end{figure}
Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a3175391e782afc092495616b4d35b91b}\label{classv8_1_1TryCatch_a3175391e782afc092495616b4d35b91b}} 
\index{v8::TryCatch@{v8::TryCatch}!HasTerminated@{HasTerminated}}
\index{HasTerminated@{HasTerminated}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{HasTerminated()}{HasTerminated()}}
{\footnotesize\ttfamily bool v8\+::\+Try\+Catch\+::\+Has\+Terminated (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const}



Returns true if an exception has been caught due to script execution being terminated. 

There is no Java\+Script representation of an execution termination exception. Such exceptions are thrown when the Terminate\+Execution methods are called to terminate a long-\/running script.

If such an exception has been thrown, Has\+Terminated will return true, indicating that it is possible to call Cancel\+Terminate\+Execution in order to continue calling into the engine. 

Definition at line 2829 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2829                                      \{}
\DoxyCodeLine{2830   \textcolor{keywordflow}{return} \mbox{\hyperlink{namespacev8_1_1anonymous__namespace_02api_8cc_03_ae7165265166edb47786472c2cf9a92de}{ToObject}}(\mbox{\hyperlink{classv8_1_1TryCatch_a4ae6a166af0336716516addc1f64fe1f}{exception\_}}) ==}
\DoxyCodeLine{2831          \mbox{\hyperlink{classv8_1_1internal_1_1ReadOnlyRoots}{i::ReadOnlyRoots}}(\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}).termination\_exception();}
\DoxyCodeLine{2832 \}}

\end{DoxyCode}


References v8\+::anonymous\+\_\+namespace\{api.\+cc\}\+::\+To\+Object().



Referenced by v8\+::\+Shell\+::\+Execute\+String(), and v8\+\_\+inspector\+::\+Injected\+Script\+::wrap\+Evaluate\+Result().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a3175391e782afc092495616b4d35b91b_cgraph}
\end{center}
\end{figure}
Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a3175391e782afc092495616b4d35b91b_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_aa4e77abd0d6790700442153cb036517f}\label{classv8_1_1TryCatch_aa4e77abd0d6790700442153cb036517f}} 
\index{v8::TryCatch@{v8::TryCatch}!IsVerbose@{IsVerbose}}
\index{IsVerbose@{IsVerbose}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{IsVerbose()}{IsVerbose()}}
{\footnotesize\ttfamily bool v8\+::\+Try\+Catch\+::\+Is\+Verbose (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const}



Returns true if verbosity is enabled. 



Definition at line 2896 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2896 \{ \textcolor{keywordflow}{return} \mbox{\hyperlink{classv8_1_1TryCatch_a3c2798d8716a14df4550bfadefdcf6a7}{is\_verbose\_}}; \}}

\end{DoxyCode}


Referenced by v8\+::internal\+::anonymous\+\_\+namespace\{isolate.\+cc\}\+::\+Predict\+Exception\+Catch\+At\+Frame().

Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_aa4e77abd0d6790700442153cb036517f_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a8aab3d2c4dabe7ea5ffc88ab22fbdc46}\label{classv8_1_1TryCatch_a8aab3d2c4dabe7ea5ffc88ab22fbdc46}} 
\index{v8::TryCatch@{v8::TryCatch}!JSStackComparableAddressPrivate@{JSStackComparableAddressPrivate}}
\index{JSStackComparableAddressPrivate@{JSStackComparableAddressPrivate}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{JSStackComparableAddressPrivate()}{JSStackComparableAddressPrivate()}}
{\footnotesize\ttfamily \mbox{\hyperlink{namespacev8_1_1internal_ab4955debfb9384e59e3d4e9fcab5b0a0}{internal\+::\+Address}} v8\+::\+Try\+Catch\+::\+JSStack\+Comparable\+Address\+Private (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}, {\ttfamily [private]}}



There are cases when the raw address of C++ \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} object cannot be used for comparisons with addresses into the JS stack. 

The cases are\+: 1) ARM, ARM64 and MIPS simulators which have separate JS stack. 2) Address sanitizer allocates local C++ object in the heap when Use\+After\+Return mode is enabled. This method returns address that can be used for comparisons with addresses into the JS stack. When neither simulator nor ASAN\textquotesingle{}s Use\+After\+Return is enabled, then the address returned will be the address of the C++ try catch handler itself. 

Definition at line 273 of file v8-\/exception.\+h.


\begin{DoxyCode}{0}
\DoxyCodeLine{273                                                     \{}
\DoxyCodeLine{274     \textcolor{keywordflow}{return} \mbox{\hyperlink{classv8_1_1TryCatch_a9c7affe9ed0b8faf49573a0709264039}{js\_stack\_comparable\_address\_}};}
\DoxyCodeLine{275   \}}

\end{DoxyCode}


Referenced by v8\+::internal\+::\+Thread\+Local\+Top\+::try\+\_\+catch\+\_\+handler\+\_\+address().

Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a8aab3d2c4dabe7ea5ffc88ab22fbdc46_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a05651b0b264e592de56f5e16bc80057f}\label{classv8_1_1TryCatch_a05651b0b264e592de56f5e16bc80057f}} 
\index{v8::TryCatch@{v8::TryCatch}!Message@{Message}}
\index{Message@{Message}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{Message()}{Message()}}
{\footnotesize\ttfamily \mbox{\hyperlink{classv8_1_1Local}{v8\+::\+Local}}$<$ \mbox{\hyperlink{classv8_1_1Message}{v8\+::\+Message}} $>$ v8\+::\+Try\+Catch\+::\+Message (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const}



Returns the message associated with this exception. 

If there is no message associated an empty handle is returned. 

Definition at line 2867 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2867                                              \{}
\DoxyCodeLine{2868   \mbox{\hyperlink{classv8_1_1internal_1_1Tagged}{i::Tagged<i::Object>}} message = \mbox{\hyperlink{namespacev8_1_1anonymous__namespace_02api_8cc_03_ae7165265166edb47786472c2cf9a92de}{ToObject}}(\mbox{\hyperlink{classv8_1_1TryCatch_a6efc81ccc783f40c660f0e709b578aaa}{message\_obj\_}});}
\DoxyCodeLine{2869   \mbox{\hyperlink{src_2base_2logging_8h_ae17f8119c108cf3070bad3449c7e0006}{DCHECK}}(IsJSMessageObject(message) || IsTheHole(message, \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}));}
\DoxyCodeLine{2870   \textcolor{keywordflow}{if} (\mbox{\hyperlink{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e}{HasCaught}}() \&\& !IsTheHole(message, \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}})) \{}
\DoxyCodeLine{2871     \textcolor{keywordflow}{return} v8::Utils::MessageToLocal(\mbox{\hyperlink{namespacev8_1_1internal_a73502f55102d18cb764881bc29f35c05}{i::direct\_handle}}(message, \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}));}
\DoxyCodeLine{2872   \} \textcolor{keywordflow}{else} \{}
\DoxyCodeLine{2873     \textcolor{keywordflow}{return} \{\};}
\DoxyCodeLine{2874   \}}
\DoxyCodeLine{2875 \}}

\end{DoxyCode}


References DCHECK, v8\+::internal\+::direct\+\_\+handle(), and v8\+::anonymous\+\_\+namespace\{api.\+cc\}\+::\+To\+Object().



Referenced by v8\+\_\+inspector\+::\+Injected\+Script\+::create\+Exception\+Details(), v8\+::internal\+::anonymous\+\_\+namespace\{liveedit.\+cc\}\+::\+Parse\+Script(), v8\+\_\+inspector\+::anonymous\+\_\+namespace\{custom-\/preview.\+cc\}\+::report\+Error(), v8\+::\+Shell\+::\+Report\+Exception(), v8\+\_\+inspector\+::\+V8\+Regex\+::\+V8\+Regex(), and v8\+\_\+inspector\+::\+Injected\+Script\+::wrap\+Evaluate\+Result().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a05651b0b264e592de56f5e16bc80057f_cgraph}
\end{center}
\end{figure}
Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a05651b0b264e592de56f5e16bc80057f_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_ad400a5dc4796e85d3d97a0e612adf0f8}\label{classv8_1_1TryCatch_ad400a5dc4796e85d3d97a0e612adf0f8}} 
\index{v8::TryCatch@{v8::TryCatch}!operator delete@{operator delete}}
\index{operator delete@{operator delete}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{operator delete()}{operator delete()}}
{\footnotesize\ttfamily void v8\+::\+Try\+Catch\+::operator delete (\begin{DoxyParamCaption}\item[{void $\ast$}]{,  }\item[{size\+\_\+t}]{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 2820 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2820 \{ \mbox{\hyperlink{classv8_1_1base_1_1OS_a3abddefa2e7ad8ab0628e6b44fad5a6b}{base::OS::Abort}}(); \}}

\end{DoxyCode}


References v8\+::base\+::\+OS\+::\+Abort().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_ad400a5dc4796e85d3d97a0e612adf0f8_cgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_aebc50debeedcab35a6a8396f73bd122a}\label{classv8_1_1TryCatch_aebc50debeedcab35a6a8396f73bd122a}} 
\index{v8::TryCatch@{v8::TryCatch}!operator delete\mbox{[}\mbox{]}@{operator delete[]}}
\index{operator delete\mbox{[}\mbox{]}@{operator delete[]}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{operator delete[]()}{operator delete[]()}}
{\footnotesize\ttfamily void v8\+::\+Try\+Catch\+::operator delete\mbox{[}$\,$\mbox{]} (\begin{DoxyParamCaption}\item[{void $\ast$}]{,  }\item[{size\+\_\+t}]{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 2821 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2821 \{ \mbox{\hyperlink{classv8_1_1base_1_1OS_a3abddefa2e7ad8ab0628e6b44fad5a6b}{base::OS::Abort}}(); \}}

\end{DoxyCode}


References v8\+::base\+::\+OS\+::\+Abort().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_aebc50debeedcab35a6a8396f73bd122a_cgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_aa6b27cdc25a9ce0806144338abd224e0}\label{classv8_1_1TryCatch_aa6b27cdc25a9ce0806144338abd224e0}} 
\index{v8::TryCatch@{v8::TryCatch}!operator new@{operator new}}
\index{operator new@{operator new}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{operator new()}{operator new()}}
{\footnotesize\ttfamily void $\ast$ v8\+::\+Try\+Catch\+::operator new (\begin{DoxyParamCaption}\item[{size\+\_\+t}]{size }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 2818 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2818 \{ \mbox{\hyperlink{classv8_1_1base_1_1OS_a3abddefa2e7ad8ab0628e6b44fad5a6b}{base::OS::Abort}}(); \}}

\end{DoxyCode}


References v8\+::base\+::\+OS\+::\+Abort().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_aa6b27cdc25a9ce0806144338abd224e0_cgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a9a9c35201c272ed3a0f9d9d6526c8637}\label{classv8_1_1TryCatch_a9a9c35201c272ed3a0f9d9d6526c8637}} 
\index{v8::TryCatch@{v8::TryCatch}!operator new\mbox{[}\mbox{]}@{operator new[]}}
\index{operator new\mbox{[}\mbox{]}@{operator new[]}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{operator new[]()}{operator new[]()}}
{\footnotesize\ttfamily void $\ast$ v8\+::\+Try\+Catch\+::operator new\mbox{[}$\,$\mbox{]} (\begin{DoxyParamCaption}\item[{size\+\_\+t}]{size }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 2819 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2819 \{ \mbox{\hyperlink{classv8_1_1base_1_1OS_a3abddefa2e7ad8ab0628e6b44fad5a6b}{base::OS::Abort}}(); \}}

\end{DoxyCode}


References v8\+::base\+::\+OS\+::\+Abort().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a9a9c35201c272ed3a0f9d9d6526c8637_cgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a9fec223509a6d7095b3088871a6084ce}\label{classv8_1_1TryCatch_a9fec223509a6d7095b3088871a6084ce}} 
\index{v8::TryCatch@{v8::TryCatch}!operator=@{operator=}}
\index{operator=@{operator=}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{operator=()}{operator=()}}
{\footnotesize\ttfamily void v8\+::\+Try\+Catch\+::operator= (\begin{DoxyParamCaption}\item[{const \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} \&}]{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [delete]}}

\mbox{\Hypertarget{classv8_1_1TryCatch_a3aae8acab4c99b374b7d782763d4c8e1}\label{classv8_1_1TryCatch_a3aae8acab4c99b374b7d782763d4c8e1}} 
\index{v8::TryCatch@{v8::TryCatch}!Reset@{Reset}}
\index{Reset@{Reset}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{Reset()}{Reset()}}
{\footnotesize\ttfamily void v8\+::\+Try\+Catch\+::\+Reset (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})}



Clears any exceptions that may have been caught by this try/catch block. 

After this method has been called, \mbox{\hyperlink{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e}{Has\+Caught()}} will return false. Cancels the scheduled exception if it is caught and \mbox{\hyperlink{classv8_1_1TryCatch_ab2cbdeeea106bcda2026b980e94c29ba}{Re\+Throw()}} is not called before.

It is not necessary to clear a try/catch block before using it again; if another exception is thrown the previously caught exception will just be overwritten. However, it is often a good idea since it makes it easier to determine which operation threw a given exception. 

Definition at line 2877 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2877                        \{}
\DoxyCodeLine{2878   \textcolor{keywordflow}{if} (\mbox{\hyperlink{classv8_1_1TryCatch_a6c8efeb878409539d0863314fba98a23}{rethrow\_}}) \textcolor{keywordflow}{return};}
\DoxyCodeLine{2879   \textcolor{keywordflow}{if} (\mbox{\hyperlink{v8config_8h_a67bd29db07c711b0fb11ff1c324ebb60}{V8\_UNLIKELY}}(\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_a5f9a69909314d298c2fd816d1310ec38}{is\_execution\_terminating}}()) \&\&}
\DoxyCodeLine{2880       !\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_aa3f53c9b9733db23efd7d26d2c854409}{thread\_local\_top}}()-\/>\mbox{\hyperlink{classv8_1_1internal_1_1ThreadLocalTop_a59794f98631d9b7aaaaf7adac9fa7fba}{CallDepthIsZero}}()) \{}
\DoxyCodeLine{2881     \textcolor{keywordflow}{return};}
\DoxyCodeLine{2882   \}}
\DoxyCodeLine{2883   \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_af2996b3f59b1763c3036a36fb0405b14}{clear\_internal\_exception}}();}
\DoxyCodeLine{2884   \mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_a3a2335d0d26d156ce3153e1237fd4075}{clear\_pending\_message}}();}
\DoxyCodeLine{2885   \mbox{\hyperlink{classv8_1_1TryCatch_ae0c8d52ffa44aaae9f9ed6aed8571aa6}{ResetInternal}}();}
\DoxyCodeLine{2886 \}}

\end{DoxyCode}


References V8\+\_\+\+UNLIKELY.



Referenced by v8\+::internal\+::anonymous\+\_\+namespace\{messages.\+cc\}\+::\+Append\+Error\+String(), v8\+::internal\+::\+Isolate\+::clear\+\_\+exception(), v8\+\_\+inspector\+::anonymous\+\_\+namespace\{value-\/mirror.\+cc\}\+::does\+Attribute\+Have\+Observable\+Side\+Effect\+On\+Get(), v8\+::internal\+::\+Error\+Utils\+::\+Format\+Stack\+Trace(), v8\+\_\+inspector\+::anonymous\+\_\+namespace\{value-\/mirror.\+cc\}\+::get\+Error\+Property(), v8\+\_\+inspector\+::\+Value\+Mirror\+::get\+Internal\+Properties(), v8\+::internal\+::\+Isolate\+::\+Report\+Exception\+Function\+Callback(), and v8\+::internal\+::\+Isolate\+::\+Report\+Exception\+Property\+Callback().

Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a3aae8acab4c99b374b7d782763d4c8e1_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_ae0c8d52ffa44aaae9f9ed6aed8571aa6}\label{classv8_1_1TryCatch_ae0c8d52ffa44aaae9f9ed6aed8571aa6}} 
\index{v8::TryCatch@{v8::TryCatch}!ResetInternal@{ResetInternal}}
\index{ResetInternal@{ResetInternal}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{ResetInternal()}{ResetInternal()}}
{\footnotesize\ttfamily void v8\+::\+Try\+Catch\+::\+Reset\+Internal (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 2888 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2888                                \{}
\DoxyCodeLine{2889   \mbox{\hyperlink{classv8_1_1internal_1_1Tagged}{i::Tagged<i::Object>}} the\_hole = \mbox{\hyperlink{classv8_1_1internal_1_1ReadOnlyRoots}{i::ReadOnlyRoots}}(\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}).the\_hole\_value();}
\DoxyCodeLine{2890   \mbox{\hyperlink{classv8_1_1TryCatch_a4ae6a166af0336716516addc1f64fe1f}{exception\_}} = \textcolor{keyword}{reinterpret\_cast<}\textcolor{keywordtype}{void}*\textcolor{keyword}{>}(the\_hole.\mbox{\hyperlink{classv8_1_1internal_1_1TaggedImpl_a4dba1b2eafcf7a414910966f950540c9}{ptr}}());}
\DoxyCodeLine{2891   \mbox{\hyperlink{classv8_1_1TryCatch_a6efc81ccc783f40c660f0e709b578aaa}{message\_obj\_}} = \textcolor{keyword}{reinterpret\_cast<}\textcolor{keywordtype}{void}*\textcolor{keyword}{>}(the\_hole.\mbox{\hyperlink{classv8_1_1internal_1_1TaggedImpl_a4dba1b2eafcf7a414910966f950540c9}{ptr}}());}
\DoxyCodeLine{2892 \}}

\end{DoxyCode}


References v8\+::internal\+::\+Tagged\+Impl$<$ k\+Ref\+Type, Storage\+Type $>$\+::ptr().



Referenced by v8\+::internal\+::\+Isolate\+::\+Cancel\+Terminate\+Execution(), and Try\+Catch().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_ae0c8d52ffa44aaae9f9ed6aed8571aa6_cgraph}
\end{center}
\end{figure}
Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_ae0c8d52ffa44aaae9f9ed6aed8571aa6_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_ab2cbdeeea106bcda2026b980e94c29ba}\label{classv8_1_1TryCatch_ab2cbdeeea106bcda2026b980e94c29ba}} 
\index{v8::TryCatch@{v8::TryCatch}!ReThrow@{ReThrow}}
\index{ReThrow@{ReThrow}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{ReThrow()}{ReThrow()}}
{\footnotesize\ttfamily \mbox{\hyperlink{classv8_1_1Local}{v8\+::\+Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{v8\+::\+Value}} $>$ v8\+::\+Try\+Catch\+::\+Re\+Throw (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})}



Throws the exception caught by this \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} in a way that avoids it being caught again by this same \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}}. 

As with Throw\+Exception it is illegal to execute any Java\+Script operations after calling Re\+Throw; the caller must return immediately to where the exception is caught. 

Definition at line 2834 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2834                                      \{}
\DoxyCodeLine{2835   \textcolor{keywordflow}{if} (!\mbox{\hyperlink{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e}{HasCaught}}()) \textcolor{keywordflow}{return} \{\};}
\DoxyCodeLine{2836   \mbox{\hyperlink{classv8_1_1TryCatch_a6c8efeb878409539d0863314fba98a23}{rethrow\_}} = \textcolor{keyword}{true};}
\DoxyCodeLine{2837   \textcolor{keywordflow}{return} \mbox{\hyperlink{namespacev8_a2ff3682899e967e165efa3460661c851}{v8::Undefined}}(\textcolor{keyword}{reinterpret\_cast<}\mbox{\hyperlink{classv8_1_1Isolate}{v8::Isolate}}*\textcolor{keyword}{>}(\mbox{\hyperlink{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}{i\_isolate\_}}));}
\DoxyCodeLine{2838 \}}

\end{DoxyCode}


References v8\+::\+Undefined().



Referenced by v8\+::internal\+::anonymous\+\_\+namespace\{gc-\/extension.\+cc\}\+::\+Parse(), v8\+\_\+inspector\+::\+V8\+Console\+::query\+Objects\+Callback(), and v8\+::\+Write\+To\+File().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=325pt]{classv8_1_1TryCatch_ab2cbdeeea106bcda2026b980e94c29ba_cgraph}
\end{center}
\end{figure}
Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_ab2cbdeeea106bcda2026b980e94c29ba_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a541b8fa6951bd5a439692c22d5c7b73c}\label{classv8_1_1TryCatch_a541b8fa6951bd5a439692c22d5c7b73c}} 
\index{v8::TryCatch@{v8::TryCatch}!SetCaptureMessage@{SetCaptureMessage}}
\index{SetCaptureMessage@{SetCaptureMessage}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{SetCaptureMessage()}{SetCaptureMessage()}}
{\footnotesize\ttfamily void v8\+::\+Try\+Catch\+::\+Set\+Capture\+Message (\begin{DoxyParamCaption}\item[{bool}]{value }\end{DoxyParamCaption})}



\mbox{\hyperlink{classv8_1_1Set}{Set}} whether or not this \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}} should capture a \mbox{\hyperlink{classv8_1_1Message}{Message}} object which holds source information about where the exception occurred. 

True by default. 

Definition at line 2898 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2898 \{ \mbox{\hyperlink{classv8_1_1TryCatch_ad1f5b0368678fa2bd0d73ac17acf976e}{capture\_message\_}} = \mbox{\hyperlink{namespacev8_1_1internal_ac1a33b5fd9af1653d346746cdb5c1ac0}{value}}; \}}

\end{DoxyCode}


References v8\+::internal\+::value.



Referenced by v8\+::internal\+::anonymous\+\_\+namespace\{messages.\+cc\}\+::\+Append\+Error\+String(), v8\+::internal\+::wasm\+::\+Wasm\+Engine\+::\+Async\+Instantiate(), v8\+::internal\+::\+Source\+Text\+Module\+::\+Evaluate(), v8\+::internal\+::\+Message\+Formatter\+::\+Format(), v8\+::internal\+::anonymous\+\_\+namespace\{execution.\+cc\}\+::\+Invoke\+With\+Try\+Catch(), and v8\+::internal\+::\+Message\+Handler\+::\+Report\+Message().

Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a541b8fa6951bd5a439692c22d5c7b73c_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a032cd889d76bd596e2616df11ced8682}\label{classv8_1_1TryCatch_a032cd889d76bd596e2616df11ced8682}} 
\index{v8::TryCatch@{v8::TryCatch}!SetVerbose@{SetVerbose}}
\index{SetVerbose@{SetVerbose}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{SetVerbose()}{SetVerbose()}}
{\footnotesize\ttfamily void v8\+::\+Try\+Catch\+::\+Set\+Verbose (\begin{DoxyParamCaption}\item[{bool}]{value }\end{DoxyParamCaption})}



\mbox{\hyperlink{classv8_1_1Set}{Set}} verbosity of the external exception handler. 

By default, exceptions that are caught by an external exception handler are not reported. Call Set\+Verbose with true on an external exception handler to have exceptions caught by the handler reported as if they were not caught. 

Definition at line 2894 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2894 \{ \mbox{\hyperlink{classv8_1_1TryCatch_a3c2798d8716a14df4550bfadefdcf6a7}{is\_verbose\_}} = \mbox{\hyperlink{namespacev8_1_1internal_ac1a33b5fd9af1653d346746cdb5c1ac0}{value}}; \}}

\end{DoxyCode}


References v8\+::internal\+::value.



Referenced by v8\+::internal\+::anonymous\+\_\+namespace\{messages.\+cc\}\+::\+Append\+Error\+String(), v8\+::internal\+::wasm\+::\+Wasm\+Engine\+::\+Async\+Instantiate(), v8\+::\+Shell\+::\+Do\+Host\+Import\+Module\+Dynamically(), v8\+::internal\+::\+Source\+Text\+Module\+::\+Evaluate(), v8\+::\+Shell\+::\+Execute\+String(), v8\+::internal\+::\+Message\+Formatter\+::\+Format(), v8\+::internal\+::anonymous\+\_\+namespace\{execution.\+cc\}\+::\+Invoke\+With\+Try\+Catch(), v8\+::\+Worker\+::\+Process\+Messages(), v8\+::anonymous\+\_\+namespace\{d8.\+cc\}\+::\+Process\+Messages(), v8\+::internal\+::\+Message\+Handler\+::\+Report\+Message(), v8\+::internal\+::\+Finalization\+Registry\+Cleanup\+Task\+::\+Run\+Internal(), v8\+::\+Inspector\+Client\+::run\+Message\+Loop\+On\+Pause(), v8\+::\+Async\+Hooks\+::\+Shell\+Promise\+Hook(), and v8\+::\+Shell\+::\+Trigger\+On\+Profile\+End\+Listener().

Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a032cd889d76bd596e2616df11ced8682_icgraph}
\end{center}
\end{figure}
\mbox{\Hypertarget{classv8_1_1TryCatch_a40e58ea4e31de973e22b2c9d86d30d57}\label{classv8_1_1TryCatch_a40e58ea4e31de973e22b2c9d86d30d57}} 
\index{v8::TryCatch@{v8::TryCatch}!StackTrace@{StackTrace}}
\index{StackTrace@{StackTrace}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{StackTrace()}{StackTrace()}\hspace{0.1cm}{\footnotesize\ttfamily [1/2]}}
{\footnotesize\ttfamily \mbox{\hyperlink{classv8_1_1MaybeLocal}{Maybe\+Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{Value}} $>$ v8\+::\+Try\+Catch\+::\+Stack\+Trace (\begin{DoxyParamCaption}\item[{\mbox{\hyperlink{classv8_1_1Local}{Local}}$<$ \mbox{\hyperlink{classv8_1_1Context}{Context}} $>$}]{context }\end{DoxyParamCaption}) const}



Returns the .stack property of the thrown object. 

If no .stack property is present or if this try/catch block has not caught an exception, an empty handle is returned. 

Definition at line 2862 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2862                                                                      \{}
\DoxyCodeLine{2863   \textcolor{keywordflow}{if} (!\mbox{\hyperlink{classv8_1_1TryCatch_ab77b36a4adf5681da59abfe43adc4e4e}{HasCaught}}()) \textcolor{keywordflow}{return} \{\};}
\DoxyCodeLine{2864   \textcolor{keywordflow}{return} \mbox{\hyperlink{classv8_1_1TryCatch_a5d80d54b34da0973f1acb30857add186}{StackTrace}}(context, \mbox{\hyperlink{classv8_1_1TryCatch_a5b86f4229e0725a3eacc69826da2e984}{Exception}}());}
\DoxyCodeLine{2865 \}}

\end{DoxyCode}
\mbox{\Hypertarget{classv8_1_1TryCatch_a5d80d54b34da0973f1acb30857add186}\label{classv8_1_1TryCatch_a5d80d54b34da0973f1acb30857add186}} 
\index{v8::TryCatch@{v8::TryCatch}!StackTrace@{StackTrace}}
\index{StackTrace@{StackTrace}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{StackTrace()}{StackTrace()}\hspace{0.1cm}{\footnotesize\ttfamily [2/2]}}
{\footnotesize\ttfamily \mbox{\hyperlink{classv8_1_1MaybeLocal}{Maybe\+Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{Value}} $>$ v8\+::\+Try\+Catch\+::\+Stack\+Trace (\begin{DoxyParamCaption}\item[{\mbox{\hyperlink{classv8_1_1Local}{Local}}$<$ \mbox{\hyperlink{classv8_1_1Context}{Context}} $>$}]{context,  }\item[{\mbox{\hyperlink{classv8_1_1Local}{Local}}$<$ \mbox{\hyperlink{classv8_1_1Value}{Value}} $>$}]{exception }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [static]}}



Returns the .stack property of an object. 

If no .stack property is present an empty handle is returned. 

Definition at line 2848 of file api.\+cc.


\begin{DoxyCode}{0}
\DoxyCodeLine{2849                                                                    \{}
\DoxyCodeLine{2850   \textcolor{keyword}{auto} i\_exception = \mbox{\hyperlink{classv8_1_1Utils_a9d3ed89b4ca4e995ba472f8091ea424f}{Utils::OpenDirectHandle}}(*exception);}
\DoxyCodeLine{2851   \textcolor{keywordflow}{if} (!IsJSObject(*i\_exception)) \textcolor{keywordflow}{return} \{\};}
\DoxyCodeLine{2852   PrepareForExecutionScope api\_scope\{context, RCCId::kAPI\_TryCatch\_StackTrace\};}
\DoxyCodeLine{2853   \mbox{\hyperlink{classv8_1_1internal_1_1Isolate}{i::Isolate}}* i\_isolate = api\_scope.i\_isolate();}
\DoxyCodeLine{2854   \textcolor{keyword}{auto} obj = i::Cast<i::JSObject>(i\_exception);}
\DoxyCodeLine{2855   \mbox{\hyperlink{classv8_1_1internal_1_1DirectHandle}{i::DirectHandle<i::String>}} \mbox{\hyperlink{flag-definitions_8h_a3ced3697c19e78b3fb4994a89f46945e}{name}} = i\_isolate-\/>\mbox{\hyperlink{classv8_1_1internal_1_1Isolate_a7e2962359009581ba84e6a361f574f61}{factory}}()-\/>stack\_string();}
\DoxyCodeLine{2856   Maybe<bool> maybe = \mbox{\hyperlink{classv8_1_1internal_1_1JSReceiver_ad0df2ef8ddcadcd2c2d7d96b19b7c3d3}{i::JSReceiver::HasProperty}}(i\_isolate, obj, \mbox{\hyperlink{flag-definitions_8h_a3ced3697c19e78b3fb4994a89f46945e}{name}});}
\DoxyCodeLine{2857   \textcolor{keywordflow}{if} (maybe != \mbox{\hyperlink{namespacev8_a41f0e280eadfba6ac9afd833fdd540cf}{Just}}(\textcolor{keyword}{true})) \textcolor{keywordflow}{return} \{\};}
\DoxyCodeLine{2858   \textcolor{keywordflow}{return} api\_scope.EscapeMaybe(}
\DoxyCodeLine{2859       \mbox{\hyperlink{classv8_1_1internal_1_1JSReceiver_aba4a2cfb4946da1635ae2730b710ff00}{i::JSReceiver::GetProperty}}(i\_isolate, obj, \mbox{\hyperlink{flag-definitions_8h_a3ced3697c19e78b3fb4994a89f46945e}{name}}));}
\DoxyCodeLine{2860 \}}

\end{DoxyCode}


References v8\+::internal\+::\+Isolate\+::factory(), v8\+::internal\+::\+JSReceiver\+::\+Get\+Property(), v8\+::internal\+::\+JSReceiver\+::\+Has\+Property(), v8\+::\+Just(), name, and v8\+::\+Utils\+::\+Open\+Direct\+Handle().



Referenced by v8\+::\+Shell\+::\+Report\+Exception().

Here is the call graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a5d80d54b34da0973f1acb30857add186_cgraph}
\end{center}
\end{figure}
Here is the caller graph for this function\+:
\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classv8_1_1TryCatch_a5d80d54b34da0973f1acb30857add186_icgraph}
\end{center}
\end{figure}


\doxysubsection{Friends And Related Function Documentation}
\mbox{\Hypertarget{classv8_1_1TryCatch_a438c8b784134d31d7a50bc60098d8ba1}\label{classv8_1_1TryCatch_a438c8b784134d31d7a50bc60098d8ba1}} 
\index{v8::TryCatch@{v8::TryCatch}!internal::Isolate@{internal::Isolate}}
\index{internal::Isolate@{internal::Isolate}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{internal::Isolate}{internal::Isolate}}
{\footnotesize\ttfamily friend class \mbox{\hyperlink{classv8_1_1internal_1_1Isolate}{internal\+::\+Isolate}}\hspace{0.3cm}{\ttfamily [friend]}}



Definition at line 289 of file v8-\/exception.\+h.

\mbox{\Hypertarget{classv8_1_1TryCatch_a97fa4264c507616255cedf7a87ec1910}\label{classv8_1_1TryCatch_a97fa4264c507616255cedf7a87ec1910}} 
\index{v8::TryCatch@{v8::TryCatch}!internal::ThreadLocalTop@{internal::ThreadLocalTop}}
\index{internal::ThreadLocalTop@{internal::ThreadLocalTop}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{internal::ThreadLocalTop}{internal::ThreadLocalTop}}
{\footnotesize\ttfamily friend class \mbox{\hyperlink{classv8_1_1internal_1_1ThreadLocalTop}{internal\+::\+Thread\+Local\+Top}}\hspace{0.3cm}{\ttfamily [friend]}}



Definition at line 290 of file v8-\/exception.\+h.



\doxysubsection{Member Data Documentation}
\mbox{\Hypertarget{classv8_1_1TryCatch_a92c1bd92eecbc728d6c2150c338fef7d}\label{classv8_1_1TryCatch_a92c1bd92eecbc728d6c2150c338fef7d}} 
\index{v8::TryCatch@{v8::TryCatch}!can\_continue\_@{can\_continue\_}}
\index{can\_continue\_@{can\_continue\_}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{can\_continue\_}{can\_continue\_}}
{\footnotesize\ttfamily bool v8\+::\+Try\+Catch\+::can\+\_\+continue\+\_\+\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 285 of file v8-\/exception.\+h.



Referenced by v8\+::internal\+::\+Isolate\+::\+Set\+Termination\+On\+External\+Try\+Catch().

\mbox{\Hypertarget{classv8_1_1TryCatch_ad1f5b0368678fa2bd0d73ac17acf976e}\label{classv8_1_1TryCatch_ad1f5b0368678fa2bd0d73ac17acf976e}} 
\index{v8::TryCatch@{v8::TryCatch}!capture\_message\_@{capture\_message\_}}
\index{capture\_message\_@{capture\_message\_}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{capture\_message\_}{capture\_message\_}}
{\footnotesize\ttfamily bool v8\+::\+Try\+Catch\+::capture\+\_\+message\+\_\+\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 286 of file v8-\/exception.\+h.



Referenced by v8\+::internal\+::\+Isolate\+::\+Throw().

\mbox{\Hypertarget{classv8_1_1TryCatch_a4ae6a166af0336716516addc1f64fe1f}\label{classv8_1_1TryCatch_a4ae6a166af0336716516addc1f64fe1f}} 
\index{v8::TryCatch@{v8::TryCatch}!exception\_@{exception\_}}
\index{exception\_@{exception\_}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{exception\_}{exception\_}}
{\footnotesize\ttfamily void$\ast$ v8\+::\+Try\+Catch\+::exception\+\_\+\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 281 of file v8-\/exception.\+h.



Referenced by v8\+::internal\+::\+Isolate\+::\+Set\+Termination\+On\+External\+Try\+Catch().

\mbox{\Hypertarget{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}\label{classv8_1_1TryCatch_ad2ecffb7f78b108cb2d933e788f5fcb1}} 
\index{v8::TryCatch@{v8::TryCatch}!i\_isolate\_@{i\_isolate\_}}
\index{i\_isolate\_@{i\_isolate\_}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{i\_isolate\_}{i\_isolate\_}}
{\footnotesize\ttfamily \mbox{\hyperlink{classv8_1_1internal_1_1Isolate}{internal\+::\+Isolate}}$\ast$ v8\+::\+Try\+Catch\+::i\+\_\+isolate\+\_\+\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 279 of file v8-\/exception.\+h.



Referenced by Try\+Catch().

\mbox{\Hypertarget{classv8_1_1TryCatch_a3c2798d8716a14df4550bfadefdcf6a7}\label{classv8_1_1TryCatch_a3c2798d8716a14df4550bfadefdcf6a7}} 
\index{v8::TryCatch@{v8::TryCatch}!is\_verbose\_@{is\_verbose\_}}
\index{is\_verbose\_@{is\_verbose\_}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{is\_verbose\_}{is\_verbose\_}}
{\footnotesize\ttfamily bool v8\+::\+Try\+Catch\+::is\+\_\+verbose\+\_\+\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 284 of file v8-\/exception.\+h.



Referenced by v8\+::internal\+::\+Isolate\+::\+Report\+Pending\+Messages(), and v8\+::internal\+::\+Isolate\+::\+Throw().

\mbox{\Hypertarget{classv8_1_1TryCatch_a9c7affe9ed0b8faf49573a0709264039}\label{classv8_1_1TryCatch_a9c7affe9ed0b8faf49573a0709264039}} 
\index{v8::TryCatch@{v8::TryCatch}!js\_stack\_comparable\_address\_@{js\_stack\_comparable\_address\_}}
\index{js\_stack\_comparable\_address\_@{js\_stack\_comparable\_address\_}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{js\_stack\_comparable\_address\_}{js\_stack\_comparable\_address\_}}
{\footnotesize\ttfamily \mbox{\hyperlink{namespacev8_1_1internal_ab4955debfb9384e59e3d4e9fcab5b0a0}{internal\+::\+Address}} v8\+::\+Try\+Catch\+::js\+\_\+stack\+\_\+comparable\+\_\+address\+\_\+\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 283 of file v8-\/exception.\+h.



Referenced by Try\+Catch().

\mbox{\Hypertarget{classv8_1_1TryCatch_a6efc81ccc783f40c660f0e709b578aaa}\label{classv8_1_1TryCatch_a6efc81ccc783f40c660f0e709b578aaa}} 
\index{v8::TryCatch@{v8::TryCatch}!message\_obj\_@{message\_obj\_}}
\index{message\_obj\_@{message\_obj\_}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{message\_obj\_}{message\_obj\_}}
{\footnotesize\ttfamily void$\ast$ v8\+::\+Try\+Catch\+::message\+\_\+obj\+\_\+\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 282 of file v8-\/exception.\+h.

\mbox{\Hypertarget{classv8_1_1TryCatch_ae1e039966ea18612bb2ecfdd2602ea84}\label{classv8_1_1TryCatch_ae1e039966ea18612bb2ecfdd2602ea84}} 
\index{v8::TryCatch@{v8::TryCatch}!next\_@{next\_}}
\index{next\_@{next\_}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{next\_}{next\_}}
{\footnotesize\ttfamily \mbox{\hyperlink{classv8_1_1TryCatch}{Try\+Catch}}$\ast$ v8\+::\+Try\+Catch\+::next\+\_\+\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 280 of file v8-\/exception.\+h.



Referenced by v8\+::internal\+::\+Isolate\+::\+Iterate(), and v8\+::internal\+::\+Isolate\+::\+Unregister\+Try\+Catch\+Handler().

\mbox{\Hypertarget{classv8_1_1TryCatch_a6c8efeb878409539d0863314fba98a23}\label{classv8_1_1TryCatch_a6c8efeb878409539d0863314fba98a23}} 
\index{v8::TryCatch@{v8::TryCatch}!rethrow\_@{rethrow\_}}
\index{rethrow\_@{rethrow\_}!v8::TryCatch@{v8::TryCatch}}
\doxysubsubsection{\texorpdfstring{rethrow\_}{rethrow\_}}
{\footnotesize\ttfamily bool v8\+::\+Try\+Catch\+::rethrow\+\_\+\hspace{0.3cm}{\ttfamily [private]}}



Definition at line 287 of file v8-\/exception.\+h.



The documentation for this class was generated from the following files\+:\begin{DoxyCompactItemize}
\item 
/mnt/\+V8\+Source\+Code/include/\mbox{\hyperlink{v8-exception_8h}{v8-\/exception.\+h}}\item 
/mnt/\+V8\+Source\+Code/src/api/\mbox{\hyperlink{api_8cc}{api.\+cc}}\end{DoxyCompactItemize}
